### ISSUE TITLE
Clarify exported span attribute precedence 

### ISSUE THREAD

**Issue by jtescher** [open-telemetry/opentelemetry-specification#660]
If a span has an attribute set which will generally also be added by an exporter (e.g. `span.kind`), currently exporters in many languages will end up with both values being reported. We should clarify the expected behavior here, e.g. should some attribute names be reserved and filtered out of the attribute set? Should both values be reported? Should the attribute-defined value be taken and the span value be ignored?

**Comment by carlosalberto** [comment]
Exporters are expected to get read-only data. Maybe you are referring to SpanProcessors?

**Comment by Oberon00** [comment]
I think this is about conventions for translating OTel span data to non-OT protocols that don't have dedicated fields for e.g. the span kind. These exporters would presumably translate the span kind into some attribute. The question now is, what should happen if an attribute with that key already exists?

**Comment by jtescher** [comment]
@Oberon00 exactly. For example should the Jaeger exporter chose the span kind or the attribute named `span.kind` if they differ?

**Comment by arminru** [comment]
Rather than defining a precedence for overwriting values I would avoid this situation beforehand by introducing _reserved_ attribute keys for all first-class citizens (span name, kind, status, ...) to the spec.

**Comment by Oberon00** [comment]
That makes a lot of sense IMHO.

**Comment by bogdandrutu** [comment]
> Rather than defining a precedence for overwriting values I would avoid this situation beforehand by introducing reserved attribute keys for all first-class citizens (span name, kind, status, ...) to the spec.

It will be hard to reserve all possible values, and specify that for any random protocol that exists, for example are we going to do the same for datadog, dynatrace, jaege, lightstep, splunk, x-ray, zipkin, etc.? Probably not, so a more scalable solution should be implemented for this.

**Comment by Oberon00** [comment]
We would not be reserving names for protocol specific fields, but for things that a span contains according to the OpenTelemetry span API specification.

**Comment by bogdandrutu** [comment]
@Oberon00 and let's say Span.Kind which is a concept in OpenTelemetry what are you going to reserve? `span.kind` why? is that not a OpenTracing convention, why not `kind`?

It will get bad very soon.

**Comment by arminru** [comment]
@bogdandrutu `"span.kind"` sounds reasonable as a reserved attribute for protocols/data models that don't support a native `Kind` of a `Span`. In this particular case, OpenTracing happens to have their own [semantic convention](https://github.com/opentracing/specification/blob/master/semantic_conventions.md) `"span.kind"` already matching OpenTelemetry's first-class `Span.Kind`, so that should be fine.
Otherwise, prefixing it with `otel` like we have for `"otel.instrumentation_library.name"` and `"otel.instrumentation_library.version"` in the [Zipkin](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/sdk_exporters/zipkin.md) and [Jaeger](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/sdk_exporters/jaeger.md) exporter spec would be safer to avoid collisions.

**Comment by svrnm** [comment]
The label removal we applied during triage stopped the `stale` bot from closing this issue, so we close it manually, if this is incorrect please re-open or raise a new issue, cc @mx-psi @jpkrohling
